### PR TITLE
Bitrot plugin code clean up

### PR DESCRIPTION
**Pull request by Madhu-1**
done code clean up where its required in bitrot plugin.
updated code for missing error handling,

if called function is returning an error,
instead of logging with the logger.WithFileds() log with the logger.Error()

updated endpoints.md

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@prashanthpai @aravindavk @rishubhjain  please review

File: plugins/bitrot/rest.go [hunk 1]
```diff
@@ -141,7 +137,8 @@ func bitrotDisableHandler(w http.ResponseWriter, r *http.Request) {
 	volinfo.Options[keyFeaturesScrub] = "false"
 
 	if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
-		logger.WithError(err).Error("failed to set volinfo in transaction context")
+		logger.WithError(err).WithField("volname",
```
Comment (rishubhjain) [inline plugins/bitrot/rest.go]: @Madhu-1 We have decided a generic message to be logged when setting a key in transaction context fails.https://github.com/gluster/glusterd2/blob/master/plugins/device/rest.go#L85.
It becomes easier to debug. Though this also works but providing the key in the WithField is helpful in debugging.

Comment (rishubhjain) [reply plugins/bitrot/rest.go]: I guess you have asked for an explanation for the same in other PRs as well.

File: doc/endpoints.md [hunk 2]
```diff
@@ -70,9 +70,7 @@ GeoreplicationSshKeyGet | GET | /ssh-key/{volname} | [](https://godoc.org/github
...
 BitrotDisable | POST | /volumes/{volname}/bitrot/disable | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#) | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#)
 ScrubOndemand | POST | /volumes/{volname}/bitrot/scrubondemand | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#) | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#)
-ScrubStatus | GET | /volumes/{volname}/bitrot/scrubstatus | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#) | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#)
-QuotaEnable | POST | /quota/{volname} | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#) | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#)
-QuotaDisable | DELETE | /quota/{volname} | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#) | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#)
+ScrubStatus | GET | /volumes/{volname}/bitrot/scrubstatus | [](https://godoc.org/github.com/gluster/glusterd2/pkg/api#) | [ScrubStatus](https://godoc.org/github.com/gluster/glusterd2/pkg/api#ScrubStatus)
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply doc/endpoints.md]: not generating endpoints as part of this PR, i will update this in upcoming PR

**Comment by Madhu-1** [comment]
@prashanthpai addressed review comments please check

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**